pandas/core/algorithms.py

```diff
@@ -944,14 +944,60 @@ def select_n_frame(frame, columns, n, method, keep):
     -------
     nordered : DataFrame
     """
-    from pandas.core.series import Series
+    from pandas import Int64Index
```

Comment:
because *some* of the routines from algorithms are imported at the very top level (e.g. ``pd.factorize`` and you would get circular imports otherwise.